### PR TITLE
Fix language tests

### PR DESCRIPTION
**Pull request by djansen-redhat**
Significant changes to language add/remove behaviour in projects.
Moves the test page behaviour to a utility class, as it is shared in
project and version settings.

### CODE REVIEW

File: functional-test/src/main/java/org/zanata/page/projects/projectsettings/ProjectLanguagesTab.java [hunk 1]
```diff
@@ -43,8 +39,10 @@
 @Slf4j
 public class ProjectLanguagesTab extends ProjectBasePage {
 
-    private By addNewLanguageField = By.id("languageAutocomplete-autocomplete__input");
-    private By settingsLanguagesForm = By.id("settings-languages-form");
+    private By activeLocales = By.id("active-locales-list");
+    private By inactiveLocales = By.id("available-locales-list");
```
Comment (davidmason) [inline functional-test/src/main/java/org/zanata/page/projects/projectsettings/ProjectLanguagesTab.java]: Would it be easier to follow if all the identifiers stuck to `enabled` and `disabled`?

Comment (davidmason) [inline functional-test/src/main/java/org/zanata/page/projects/projectsettings/ProjectLanguagesTab.java]: At this point I realize that the `id` of the components doesn't help. Part of cleaning up that code will be to make all those consistently use `enabled` and `disabled` as well.

File: functional-test/src/main/java/org/zanata/page/projects/projectsettings/ProjectLanguagesTab.java [hunk 2]
```diff
@@ -111,37 +119,24 @@ public ProjectLanguagesTab enterSearchLanguage(String languageQuery) {
      * Add a language to the languages list.
      * Assumes that the search string has already been entered.
```
Comment (davidmason) [inline functional-test/src/main/java/org/zanata/page/projects/projectsettings/ProjectLanguagesTab.java]: Does this comment about the search string still count, now that this does not use an autocomplete component?

File: functional-test/src/main/java/org/zanata/page/projects/projectsettings/ProjectLanguagesTab.java [hunk 3]
```diff
@@ -111,37 +119,24 @@ public ProjectLanguagesTab enterSearchLanguage(String languageQuery) {
...
-                }
-                return clickedLocale;
+                return LanguageList.toggleLanguageInList(
+                        getDriver().findElement(inactiveLocales),
+                        searchLocaleId);
             }
         });
-
-
         refreshPageUntil(this, new Predicate<WebDriver>() {
```
Comment (davidmason) [inline functional-test/src/main/java/org/zanata/page/projects/projectsettings/ProjectLanguagesTab.java]: This looks like the equivalent of pressing F5 repeatedly (going by the method name). Is that right?

Comment (davidmason) [inline functional-test/src/main/java/org/zanata/page/projects/projectsettings/ProjectLanguagesTab.java]: Or does it just refresh selenium's model of the page?

Comment (djansen-redhat) [inline functional-test/src/main/java/org/zanata/page/projects/projectsettings/ProjectLanguagesTab.java]: No. Yes.

File: functional-test/src/main/java/org/zanata/page/projects/projectsettings/ProjectLanguagesTab.java (hunk 2 above)
Comment (djansen-redhat) [inline functional-test/src/main/java/org/zanata/page/projects/projectsettings/ProjectLanguagesTab.java]: Fixed.

**Comment by davidmason** [comment]
:+1:   :sake:

**Pull request merged**